manifests/init.pp

```diff
@@ -7,9 +7,9 @@
   Hash $common_config = {},
   String $libdir,
   String $configdir,
-  String $plugin_owner,
-  String $plugin_group,
-  String $plugin_mode,
+  Optional[String] $plugin_owner,
+  Optional[String] $plugin_group,
+  Optional[String] $plugin_mode,
```

Comment:
wouldnt't what you did in windows.yaml - set them to `~` - be enough? they would be undef on windows and so the conditionals added below to `init.pp` would be not needed?